folktables/utils/download_utils.py

```diff
@@ -0,0 +1,135 @@
+import concurrent.futures
+import os
+import pathlib
+
+import requests
+
+from folktables import exceptions
+
+
+def download_file(url, download_path):
+    """Makes a GET request to the specified URL and saves the
+    contents of the response to the specified path.
+
+    Parameters
+    ----------
+    url : str
+        URL from where the data will be downloaded.
+    download_path : pathlib.Path
+        Path to where the downloaded content will be stored.
+
+    Returns
+    -------
+    download_path : str
+        Path to where the content is stored.
+
+    Raises
+    ------
+    exceptions.FileDownloadError
+        This exception is raised if we get an error from our HTTP request
+        (i.e., the status code we get from the response is not 200).
+    """
+    response = requests.get(url)
+
+    if response.status_code != 200:
+        raise exceptions.FileDownloadError(
+            f'Failed to download the data from: {url}\n'
+            f'The HTTP request returned a {response.status_code} status code.'
+        )
+
+    with open(download_path, 'wb') as handle:
+        handle.write(response.content)
+
+    return download_path
+
+
+def async_download_file(files_resource):
```

Comment:
The name is maybe a bit confusing because the download call in this function is actually synchronous, right? If the only place we're using this function is Line 131 below, maybe the more elegant approach is to use a lambda function wrapper around `download_file` in Line 131 instead?

Reply:
Good point and I like the idea of using a lambda function wrapper. I'll go ahead and update the code.